From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoftikLord <dimap9986@gmail.com>
Date: Sun, 9 Apr 2023 04:37:30 +0500
Subject: [PATCH] 
 Paper-limit-the-range-at-which-we-ll-consider-an-attackable-target


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 76da61fc6298a16476a15f9f150a82708fb3d13c..db1ea3f40ca9d5681c4f42315fd9b96fbd7def6a 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -2944,9 +2944,13 @@ public abstract class World implements IBlockAccess {
 
         for (int i = 0; i < this.players.size(); ++i) {
             EntityHuman entityhuman1 = (EntityHuman) this.players.get(i);
-
+            // Paper start
+            // move distance check up, if set, check distance^2 is less than XZlimit^2, continue
+            // 4th method param is XZlimit (at least at the time of commit)
+            double d6 = entityhuman1.d(d0, entityhuman1.locY, d2);
+            if (d3 < 0.0D || d6 < d3 * d3)
             if (!entityhuman1.abilities.isInvulnerable && entityhuman1.isAlive() && !entityhuman1.isSpectator() && (predicate == null || predicate.apply(entityhuman1))) {
-                double d6 = entityhuman1.d(d0, entityhuman1.locY, d2);
+                // Paper end
                 double d7 = d3;
 
                 if (entityhuman1.isSneaking()) {
